modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java

```diff
@@ -169,6 +179,57 @@ public AppSchemaDataAccess(Set<FeatureTypeMapping> mappings, boolean hidden)
         register();
     }
 
+    /**
+     * Checks if the mapping is an include and is identical to the previous one.
+     *
+     * @param mapping the mapping to check
+     * @return true if the mapping is an include and is identical to the previous one
+     */
+    private boolean isIncludeAndDeepSame(FeatureTypeMapping mapping) {
+        // Not an include, so no need to check
+        if (!Boolean.TRUE.equals(mapping.isInclude())) {
+            return false;
+        }
+        FeatureTypeMapping compareMapping = null;
+        // Get the mapping to compare with
+        if (mapping.getMappingName() != null) {
+            Name name = mapping.getMappingName();
+            if (this.mappings.containsKey(name)) {
+                compareMapping = this.mappings.get(name);
+            } else {
+                try {
+                    if (DataAccessRegistry.hasName(name)) {
+                        AppSchemaDataAccess appSchemaDataAccess =
+                                (AppSchemaDataAccess) DataAccessRegistry.getDataAccess(name);
+                        compareMapping = appSchemaDataAccess.getMappingByName(name);
+                    }
+                } catch (IOException | ClassCastException e) {
+                    LOGGER.fine(
+                            "Could not get mapping from registry for "
+                                    + name
+                                    + " while trying to test for duplicates");
+                    return false;
+                }
+            }
```

Comment:
I'm uncertain about this aspect, but if my understanding is accurate, the objective is to prevent App-Schema from raising errors about duplicate mappings when the exact same feature type mapping is encountered multiple times due to being included by different mappings.

It appears that we're conducting a search in two different places:

1. Firstly, in the mappings hashmap of this `AppSchemaDataAccess` instance. I presume that an `AppSchemaDataAccess` instance is associated with a root mapping in App-Schema.
2. Secondly, in the `DataAccessRegistry` registry, which encompasses all the root mappings registered in App-Schema.

The search for duplicates encompasses both:

```java
if (this.mappings.containsKey(name) || DataAccessRegistry.hasName(name)) {
```

I would suggest adding comments in the code to elucidate the process. It seems that the initial check is performed within the current mappings, as if the mapping is found there, it implies it has already been validated against the common registry.

Reply:
Comments added.